### PR TITLE
fix: metal on conda on a13 and higher

### PR DESCRIPTION
**Pull request by Leikoe**
metal not compiling successfully on conda is due to a wrong compile requestType. This can be fixed the following ways:
- compile with a fairly recent libSystem.B.dylib
- compiling a dynamic lib
- patch Metal
- compiling a executable library (as we do right now) + linking to any dynamic libary

This fix is using the last way. It will work on any mac which supports dynamic library and will only trigger then. For older devices it won't change anything.

### CODE REVIEW

**Comment by geohot** [comment]
Bad spacing. Don't use globals.

File: tinygrad/runtime/ops_metal.py [hunk 1]
```diff
@@ -11,10 +11,20 @@ def wait_check(cbuf: Any):
   if (error := cbuf.error()) is not None:
     raise RuntimeError(error)
 
+def get_workaround_lib(device):
+    options = Metal.MTLCompileOptions.new()
```
Comment (chenyuxyz) [inline tinygrad/runtime/ops_metal.py:15]: bad indent-width

**Comment by geohot** [comment]
Can we submit this request directly?

**Comment by Leikoe** [comment]
No, I looked for other ways but apple expects you not to use libraryDataContents and therefore store incomplete library data on some versions. The only way to do it properly is either to recompile and pray for system cache or use things with very limited support (BinaryArchive).

**Comment by Leikoe** [comment]
If you were to call private C++ apis from Metal internals you could send the proper requestType but that wouldn't be supported everywhere, this hack is not as bad.

**Comment by geohot** [comment]
Not going to merge a hack like this. From what I gather it really is a conda bug and should be filed as an issue there.

**Pull request closed without merging**